File: editor/en/[2433]Find The Original Array of Prefix Xor.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int[] findArray(int[] pref) {
        int[] arr = new int[pref.length];

        arr[0] = pref[0];

        for (int i = 1; i < pref.length; i++) {
            arr[i] = pref[i] ^ pref[i - 1];
        }

        return arr;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[2433]Find The Original Array of Prefix Xor.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} pref
 * @return {number[]}
 */
var findArray = function (pref) {
    let arr = [pref[0]];

    for (let i = 1; i < pref.length; i++) {
        arr[i] = pref[i] ^ pref[i - 1];
    }

    return arr;
};
//leetcode submit region end(Prohibit modification and deletion)
